atoms/auth.js

```diff
@@ -0,0 +1,84 @@
+import { atom, selector } from 'recoil';
+
+export const firstName = atom({
+  key: 'auth-firstName',
+  default: '',
+});
+
+export const lastName = atom({
+  key: 'auth-lastName',
+  default: '',
+});
+
+export const email = atom({
+  key: 'auth-email',
+  default: '',
+});
+
+export const password = atom({
+  key: 'auth-password',
+  default: '',
+});
+
+export const submitted = atom({
+  key: 'auth-submitted',
+  default: false,
+});
+
+export const validFirstName = selector({
+  key: 'auto-validFirstName',
```

Comment:
Just looking at recoil, it looks very easy to understand. 
A question about naming. Do the prepended `auth-` and `auto-` in the keys define anything special, or was it just to name by group.

Reply:
ahah that's an error. It should be `auth` not auto. I did that so we don't accidentally write the same key in another atom somewhere else since all keys need to be unique.